csp_adapter_symphony/adapter.py

```diff
@@ -337,109 +416,49 @@ def stop(self):
         if self._running:
             # in order to unblock current requests.get, send a message to one of the rooms we are listening on
             self._running = False
-            self._delete_datafeed_if_set()
-            if self._exit_msg:
-                send_symphony_message(self._exit_msg, next(iter(self._room_ids)), self._message_create_url, self._header)
+            try:
+                self._delete_datafeed_if_set()
+                if self._exit_msg:
+                    send_symphony_message(self._exit_msg, next(iter(self._room_ids)), self._message_create_url, self._header)
+            except Exception:
+                log.error("Error on sending exit message and deleting datafeed on shutdown", exc_info=True)
             self._thread.join()
 
+    def _get_new_ack_id_and_messages(self, ack_id: str) -> Tuple[str, List[SymphonyMessage]]:
+        ret = []
+        resp = requests.post(url=self._url, headers=self._header, json={"ackId": ack_id})
+        if resp.status_code == 400:
+            # Bad datafeed, we need a new one
+            self._set_new_datafeed()
+            return "", []
+
+        elif resp.status_code == 200:
+            msg_json = resp.json()
+            if "ackId" in msg_json:
+                ack_id = msg_json["ackId"]
+            events = msg_json.get("events", [])
+            for m in events:
+                maybe_msg = _handle_event(m, self._room_ids, self._room_mapper)
+                if maybe_msg is not None:
+                    ret.append(maybe_msg)
+        return ack_id, ret
+
     def _run(self):
         ack_id = ""
+        get_new_messages_func = (
+            self._retry_decorator(self._get_new_ack_id_and_messages) if self._retry_decorator is not None else self._get_new_ack_id_and_messages
+        )
         while self._running:
             try:
-                resp = requests.post(url=self._url, headers=self._header, json={"ackId": ack_id})
-            except Exception as e:
-                error_msg = "An exception occured trying read from the datafeed, Symphony reader shutting down..."
-                log.error(error_msg, exc_info=True)
+                ack_id, ret = get_new_messages_func(ack_id)
+            except Exception as exc:
+                # On the final failure, this happens
+                # No need to retry these calls, we are failing anyways
+                error_msg = "An exception occured trying to interact with datafeed, max_attempts exceeded. Symphony Reader is shutting down..."
+                log.error(error_msg)
```

Comment:
`log.exception`

Reply:
Here I don't want to add the traceback since we raise the exception right after anyways